### DISCUSSION TITLE
Delete Fails - Table doesn't exist, select and update work

### DISCUSSION THREAD

**Discussion by fotoflo** [planetscale/discussion#607] [Bug Reports]
hi i can select and update from a table but i cant delet from it - it says the doesn't exist?

Operation failed: There was an error while applying the SQL script to the database.
Executing:
DELETE FROM `fastmonitor-prod`.`User` WHERE (`id` = 'clous9wo3004vl30fqhmqq7n1');

ERROR 1146: 1146: target: fastmonitor-prod.-.primary: vttablet: rpc error: code = NotFound desc = Table 'fastmonitor-prod._167d1ae3_aed2_59c1_994b_4336fa6fba03_20240208133147_vrepl' doesn't exist (errno 1146) (sqlstate 42S02) (CallerID: ejot029rc3g54vxkbwjd): Sql: "delete from _167d1ae3_aed2_59c1_994b_4336fa6fba03_20240208133147_vrepl where (userId) in ::fkc_vals", BindVars: {REDACTED}
SQL Statement:
DELETE FROM `fastmonitor-prod`.`User` WHERE (`id` = 'clous9wo3004vl30fqhmqq7n1')

![image](https://github.com/planetscale/discussion/assets/1642357/d2d86c61-b16a-43cc-a325-a99be6eed6e0)

**Comment by mscoutermarsh** [comment]
Hi @fotoflo,

Are you using foreign key constraints? This looks like a bug we are currently fixing for the foreign key constraint beta. If you write into support they'll be able to better help you.

**Reply by fotoflo** [reply]
Yes i am, thanks @mscoutermarsh

**Comment by fotoflo** [comment]
hi @mscoutermarsh I emailed support two days ago without response. This is a show stopper for my app btw.

Also new deploy request seemed blocked as well
"Refusing to apply an empty vschema to a non-empty existing vschema"

**Reply by mscoutermarsh** [reply]
Sorry @fotoflo! It should be soon now that it's Monday. We don't have standard support coverage over the weekends.

**Reply by mscoutermarsh** [reply]
@fotoflo I was able to take a look just now. Is your issue with deleting data now resolved? I can see you dropped the FK constraints.

We had a bug that impacted databases with a `-` in the name. It impacted FK constraints after a deploy request, where the FK relationship was not updated to the new table.

Which is why you were seeing the error: `Table 'fastmonitor-prod._167d1ae3_aed2_59c1_994b_4336fa6fba03_20240208133147_vrepl' doesn't exist`. That is referring to a temporary table during the deploy request.

We have since fixed the bug and your database has been updated. Future deploy requests with FK constraints will work correctly.

As for your issue with: `"Refusing to apply an empty vschema to a non-empty existing vschema"`. This is due to toggling the FK beta on and off. If you delete your dev branch and create a new one, future deploy requests should work correctly.
